main.cpp

```diff
@@ -213,191 +334,379 @@ struct Computer
+    }
+    else
+    {
+        tuning -= tuningDelta;
+    }
+}
+
+void Tuner::lightUp()
+{
+    std::cout << "Dial lit up!" << std::endl;
+}
+
 struct Speaker
 {
-    // bass driver voltage
     double bassDriverVoltageVolts = 31.43;
-    // bass cone displacement
     float bassConeDisplacementMillimeters = 2.3f;
-    // treeble driver voltage
     double treebleDriverVoltageVolts = 30.56;
-    // treeble cone displacement
     float treebleConeDisplacementMillimeters = 0.4f;
-    // vent exhaust pressure
     float ventExhaustOverPressurePa = 45.121f;
+    float maxCurrent = 1.2f;
 
-    // 3 things it can do:
-    // process signal
-    float processSignal(float inputSignal); // returns a driver current 
+    float processSignal(float inputSignal); 
   
-    // emit sound
-    void emitSound(float driverCurrent);
+    float emitSound(float driverCurrent);
     
-    // blow
-    void blow(float driverCurrent);
+    bool blow(float driverCurrent);
 };
 
+float Speaker::processSignal(float inputSignal)
+{
+    return inputSignal * static_cast<float>(rand()) / static_cast<float>(RAND_MAX);
+}
+
+float Speaker::emitSound(float driverCurrent)
+{
+    float conversionConstant = 0.12842f;
+    
+    return driverCurrent*conversionConstant;
+}
+
+bool Speaker::blow(float driverCurrent)
+{
+    bool isBlown = false;
+    
+    if (std::abs(driverCurrent) > maxCurrent)
+    {
+        isBlown = true;
+    }
+
+    return isBlown;
+}
+
 struct Stereo
 {
-    // turntable
     Turntable turntable;
-    // cassette deck
     CassetteDeck cassetteDeck;
-    // amplifier
     Amplifier amplifier;
-    // tuner
     Tuner tuner;
-    // speakers
     Speaker speaker1, speaker2;
 
-    // 3 things it can do:
-    // play music
-    bool playMusic(std::string source); // returns true if music can be played from source
+    bool playMusic(std::string source);
 
-    // change radio channel
-    bool changeRadioChannel(std::string newRadioChannel); // returns true if change is successful
+    void changeRadioChannel(std::string newRadioChannel, bool up);
 
-    // record tape
-    bool recordTape(std::string inputSource); // returns true if the source can be recorded
+    bool recordTape(std::string inputSource); 
 };
 
+bool Stereo::playMusic(std::string source)
+{
+    bool isPlaying = false;
+    
+    if (source == "Turntable" || source == "Cassette" || source == "Tuner")
+    {
+        isPlaying = true;
+    }
+
+    return isPlaying;
+}
+
+void Stereo::changeRadioChannel(std::string newRadioChannel, bool up)
+{
+    std::cout << "Channel changed to: " << newRadioChannel << std::endl;
+    tuner.changeTuning(up);
+}
+
+bool Stereo::recordTape(std::string inputSource)
+{
+    bool isRecording = false;
+
+    if (inputSource == "Turntable" || inputSource == "Tuner")
+    {
+        isRecording = true;
+        cassetteDeck.rotateReel();
+    }
+
+    return isRecording;
+}
+
 int main()
 {
+    srand( static_cast<unsigned int> (time(nullptr)) );
```

Comment:
Casting has not been covered yet.  
Remove this cast.
Change your code so that no warnings are generated **without** using casting.
- [ ] revise